### PR TITLE
Add rule violations from Yamllint

### PR DESCRIPTION
**Pull request by ssbarnea**
This detect yamllint rule violations when this is found as installed. This will allow us to remove a set of rules that
overlapped with general YAML syntax ones.

Fixes: #953
Release-Notes: https://github.com/ansible-community/ansible-lint/discussions/1150

### CODE REVIEW

File: lib/ansiblelint/rules/__init__.py
Comment (greg-hellings) [inline lib/ansiblelint/rules/__init__.py:164]: Why does it look like you've added this twice?

**Comment by felixfontein** [comment]
I've tried this PR in a [role-only collection of mine](https://github.com/felixfontein/ansible-acme/), and found it ignores my .yamllint file.

**Comment by ssbarnea** [comment]
@felixfontein Thanks for looking at it. Yep, I am aware that it ignores the in-repo yamllint config and that it uses a hardcoded configuration.

That is more or less by design as I wanted to fully control the results. I can see how this could get me into trouble and I guess I will have to make it use the embedded default only when a local config is not found.

One thing that you likely already know is that most Ansible content is not compatible with the yamllint-own-default-config and that the author has very strong opinions regarding what defaults should be.

Another feature that is currently missing is ability to fully disable this rule instead of ignoring or skipping it.

I do not expect to merge this very soon but getting early feedback is essential, as it would allow me to improve the implementation.

**Comment by felixfontein** [comment]
@ssbarnea being able to disable that would be really helpful :) Does it make sense to make this configurable? I don't know if ansible-lint has other rules that can be configured (and not just disabled).

**Comment by ssbarnea** [comment]
> @ssbarnea being able to disable that would be really helpful :) Does it make sense to make this configurable? I don't know if ansible-lint has other rules that can be configured (and not just disabled).

Now is configurable, it will respect user defined config. You can also skip entire rules with "YAML" tag, or specific ones with their named-tag reported by yamllint itself.

AFAIK, it is ready to be merged. I am just waiting for an approval. Once merged I will tag a pre-release in order to allow users to start consuming the newer changes.

File: .ansible-lint
Comment (felixfontein) [inline .ansible-lint:2]: This looks a lot like a yamllint config. Who is reading and processing this part (both `extends` and `rules`)?

Comment (ssbarnea) [reply .ansible-lint]: @felixfontein Thanks, that was added by mistake, I removed it.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@ssbarnea thanks for implementing this!

**Comment by ssbarnea** [comment]
@felixfontein Thanks for reviewing it. I also have https://github.com/ansible-community/ansible-lint/pull/950/files ready for review, which is kinda similar: making use of ansible syntax check.

File: lib/ansiblelint/rules/YamllintRule.py
Comment (MarkusTeufelberger) [inline lib/ansiblelint/rules/YamllintRule.py:35]: Just to be sure: This is supposed to with `YAML` in the skip_list, not with `yaml`? To me it seems like the other way around?

Comment (ssbarnea) [reply lib/ansiblelint/rules/YamllintRule.py:35]: That is an error, correct key is 'yaml', I will fix it.

Comment (ssbarnea) [reply lib/ansiblelint/rules/YamllintRule.py:35]: Fixing via https://github.com/ansible-community/ansible-lint/pull/1335

**Pull request merged**
